### DISCUSSION TITLE
Unsure when to set server domain certificate

### DISCUSSION THREAD

**Discussion by Puetz** [Dokploy/dokploy#652] [Q&A] [answered]
Hey,

I just started using dokploy and my domain is hosted by Cloudflare.
I found a contradiction within your documentation as well as between the documenation and the [youtube video](https://www.youtube.com/watch?v=mznYKPvhcfw) which is featured on your main website [dokploy.com](https://dokploy.com/).

In the [Dokploy - Introduction](https://www.youtube.com/watch?v=mznYKPvhcfw) video the speaker is using Cloudflare and at 4:23 sets "Let's encrypt" for the server domain. When I did that, I was unable to access my dokploy website via my domain and got an error "Too many redirects".  
At the start of the [Cerfiticates documentation](https://docs.dokploy.com/en/docs/core/domain/certificates) there is a little remark:
> When you register a domain on Cloudflare, the SSL certificate is automatically assigned by Cloudflare, so you need to select None in Dokploy and disable HTTPS.

But when you scroll down there is an entire [Cloudflare Setup](https://docs.dokploy.com/en/docs/core/domain/certificates#cloudflare-setup) section saying that you should use "Let's encrypt".

For me as well as the person in this [discussion](https://github.com/Dokploy/dokploy/discussions/148) the setup only worked when using no server certificate. 

Can you please explain why it's working in the video but not for me? My DNS Setup is set to "Full" just like in the video and mentioned in the documentation.

**If** it is correct to use no certificate for domains managed by Cloudflare, does that
a) also apply to my projects and apps when the domain is managed by Cloudflare?
b) mean I should set the certificate for all projects or apps where the domain comes from a different provider?

Thanks a lot for your help!
Cheers,
Thomas

**Comment by Siumauricio** [comment, marked as answer]
Yes, there are 2 ways, I will resume like this:

Cloudflare have multiple modes of SSL:

1. If you have flexible mode, in dokploy when you create a domain you need to do -> disable HTTPS and Select none in certificate
2. if you have Full Strict mode , in dokploy when you create a domain -> enable HTTP and Select Letsencrypt in certificate

I also recommend that every time you try to bind a domain, first point the domain to the server IP before creating the domain, especially when you have Full Strict mode because if you have this mode and you create the domain first and then point the IP, what will happen is that traefik will try to generate the certificate but as it does not find records it will never generate the certificate unless you delete the domain and recreate it or restart traefik, and keep in mind that if you try too many times about 5-6 times letsencrypt will probably block you for 24 hours so make sure you always point the IP first and then create the domain.
